src/shared/settingsConfiguration.ts

```diff
@@ -139,4 +156,46 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
             throw Error(`config: unknown aws.suppressPrompts item: "${name}"`)
         }
     }
+
+    public readDevSetting<T>(key: AwsDevSetting): string
+    public readDevSetting<T>(key: AwsDevSetting, type: jsPrimitiveTypeName, silent: boolean): T | undefined
+
+    /**
+     * Gets the value of a developer only setting.
+     *
+     * TODO: show a dialog if this is used, and/or make a UI change (such as
+     * changing the AWS Explorer color) so that it's obvious that Toolkit is in
+     * "Developer mode". Throw an error in CI.
+     */
+    public readDevSetting<T>(
+        key: AwsDevSetting,
+        type: jsPrimitiveTypeName = 'string',
+        silent: boolean = false
+    ): T | undefined {
+        const config = vscode.workspace.getConfiguration()
+        const val = config.get<T>(key)
+        if (!val) {
```

Comment:
```suggestion
        if (val === undefined) {
```
Since some falsey values are potentially valid (e.g. 0)